src/trio/_core/_run.py

```diff
@@ -70,19 +76,25 @@
     # for some strange reason Sphinx works with outcome.Outcome, but not Outcome, in
     # start_guest_run. Same with types.FrameType in iter_await_frames
     import outcome
-    from typing_extensions import Self
+    from typing_extensions import Self, TypeVarTuple, Unpack
+
+    PosArgT = TypeVarTuple("PosArgT")
+
+    # Needs to be guarded, since Unpack[] would be evaluated at runtime.
+    class _NurseryStartFunc(Protocol[Unpack[PosArgT], StatusT_co]):
+        """Type of functions passed to `nursery.start() <trio.Nursery.start>`."""
+
+        def __call__(
+            self, *args: Unpack[PosArgT], task_status: TaskStatus[StatusT_co]
```

Comment:
```suggestion
            self, *args: Unpack[PosArgT], *, task_status: TaskStatus[StatusT_co]
```

This should fix the mypy errors. This actually makes sense!

Reply:
Unfortunately no, the `*args` already makes it keyword-only.

Reply:
Yeah sorry about the repeated confusion with this thread, `TypeVarTuple` is getting to my head :(

Reply:
Yeah it's real confusing. Since `task_status` is keyword-only it shouldn't get mixed up. As you can see with the type-tests, if it's being assigned to an annotated type, Mypy can do bidirectional inference and figure it out. But if it's not annotated or just a bare call, it gets confused and assumes it's `Never`. I think it's a bug.

Reply:
I agree. I took a look at mypy's code for the last 45 minutes and spotted:
 1) It looks like it isn't generating constraints for the typevar.
 2) This is because the constraints only get generated for a callback protocol if the type being assigned is a subtype of the callback protocol w/ erased tvars (I assume this is to make less confusing error messages). Unfortunately, `TypeVarTuple` gets erased to `tuple[Any, ...]` and our callable doesn't take in infinite parameters.
 3) Even if that problem were to be solved, it looks like mypy only generates constraints for either a bunch of type vars or a single unpack. This isn't a problem normally (`Callable[[Unpack[Ts], T], None]` has an implicit `Unpack`) but means that mypy focuses on the typevartuple, not making a constraint for our type variable.

I'm not really sure how to solve this myself, a bug report would probably work nicely.

Reply:
@A5rocks probably worth cross-posting your analysis in the issue that teamspen opened.